configure.ml

```diff
@@ -107,7 +107,7 @@ let run ?(fatal=true) ?(err=StdErr) prog args =
       let cmd = String.concat " " (prog::args) in
       let exn = match e with Failure s -> s | _ -> Printexc.to_string e in
       let msg = sprintf "Error while running '%s' (%s)" cmd exn in
-      if fatal then die msg else (printf "W: %s\n" msg; "", [])
+      if fatal then die msg else (printf "W: %s\n%!" msg; "", [])
```

Comment:
What are all these new `%!`. Is this related to this PR?

Reply:
That adds a flush to the formatter, which is usually necessary for printing to succeed. A better strategy would be to define a `cprintf` function that takes care of this:
```
let cprintf fmt = fprintf (fun fmt -> fprintf fmt "@\n%!") fmt
```
so now on every call of `cprintf "foo"`, the newline and flush are automatically inserted.

Reply:
Yep, I need to separate this out from this PR. I added it because some messages wouldn't show up when testing the dependency checks introduced by this PR.

Reply:
@ejgallego your `cprintf` function doesn't seem to typecheck. What did you mean there?

Reply:
@ejgallego ping

Reply:
Sorry there was a typo it should be:
```ocaml
let cprintf fmt = kfprintf (fun fmt -> fprintf fmt "@\n%!") fmt
```

Reply:
Note the eta expansion is usually mandatory.

Reply:
@ejgallego I'm still a bit lost. Why is the output channel called `fmt`?

Reply:
A mistake, I guess, so the eta-expansion is not needed.

Reply:
Oh no sorry, I see now. I was confused about a formatter for an output channel vs the output channel itself. Thanks, will split this part out of this PR.

Reply:
> @ejgallego I'm still a bit lost. Why is the output channel called fmt?

It is not an output channel but a formatter.

> A mistake, I guess, so the eta-expansion is not needed.

?? Eta expansion is needed if you want your code to work in older OCamls.

Reply:
I understand the quiproquo now. `configure.ml` uses `Printf`, not `Format`. In particular `Printf.kfprintf` takes an `output_channel`, not a `formatter`.

Reply:
Oh, is it still using `Printf`? I suggest you switch it to use `Format`.

Reply:
I'll leave that for another time, it is critical to get this sphinx infra PR ready.